kasa/cli.py

```diff
@@ -158,13 +176,17 @@ async def cli(
         return
 
     # If JSON output is requested, disable echo
+    global echo
     if json:
-        global echo
 
         def _nop_echo(*args, **kwargs):
             pass
 
         echo = _nop_echo
+    else:
+        # Set back to default is required if running tests with CliRunner
+        global _do_echo
+        echo = _do_echo
```

Comment:
Could we avoid changes to echo handling by simply overriding it inside the `test_cli.py`? That would be the preferred way, imo.